torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -606,6 +607,20 @@ class TORCH_CUDA_CU_API GridWelford final : public Expr {
   ParallelTypeBitmap thread_predicate_;
 };
 
+class TORCH_CUDA_CU_API AllocateFusedReduction final : public Expr {
```

Comment:
I added a new expression just to use the new reduction class in the runtime code. 

Instead of adding a new expression type just for this particular case, I also considered reusing `kir::Allocate`, but the data type is not any of the ones expressible with `DataType`. If it were possible to have something like a named type, it would have been possible to do something like `create<kir::Allocate>(create<NamedSymbol>("reduction"), NamedType("ParallelReduce<0, 3, 3, ....>"))`. 

If we would want to use more of these custom types, we may want to consider supporting named types.